src/routes/movies.ts

```diff
@@ -0,0 +1,191 @@
+    res.status(200).json(movie);
+  } catch (error: any) {
+    res.status(500).json({ message: error.message });
+  }
+});
+
+/**
+ * @swagger
+ * /movies/{id}:
+ *   put:
+ *     summary: Update a movie by ID
+ *     tags:
+ *       - Movies
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         required: true
+ *         description: ID of the movie to update
+ *         schema:
+ *           type: string
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/MovieInput'
+ *     responses:
+ *       200:
+ *         description: Movie updated successfully
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/Movie'
+ *       404:
+ *         description: Movie not found
+ *       500:
+ *         description: Internal server error
+ */
+
+router.put('/:id', async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params;
+    const movie = await Movie.findByIdAndUpdate(id, req.body);
+
+    if (!movie) {
+      return res.status(404).json({ message: `cannot find any movie with ID ${id}` });
+    }
+    const updatedMovie = await Movie.findById(id);
+    res.status(200).json(updatedMovie);
+  } catch (error: any) {
+    res.status(500).json({ message: error.message });
+  }
+});
+
+/**
+ * @swagger
+ * /movies/{id}:
+ *   delete:
+ *     summary: Delete a movie by ID
+ *     tags:
+ *       - Movies
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         required: true
+ *         description: ID of the movie to delete
+ *         schema:
+ *           type: string
+ *     responses:
+ *       200:
+ *         description: Movie deleted successfully
+ *       404:
+ *         description: Movie not found
+ *       500:
+ *         description: Internal server error
+ */
+
+router.delete('/:id', async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params;
+    const movie = await Movie.findByIdAndDelete(id);
+
+    if (!movie) {
+      return res.status(404).json({ message: `cannot find any movie with ID ${id}` });
+    }
+    res.status(200).json({ message: 'movie has been deleted' });
+  } catch (error: any) {
+    res.status(500).json({ message: error.message });
+  }
+});
+
+/**
+ * @swagger
+ * /movies/genre/{genreName}:
+ *   get:
+ *     summary: Get movies by genre
+ *     tags:
+ *       - Movies
+ *     parameters:
+ *       - in: path
+ *         name: genreName
+ *         required: true
+ *         description: Name of the genre to filter movies
+ *         schema:
+ *           type: string
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 $ref: '#/components/schemas/Movie'
```

Comment:
Receive wrong succesfull payload response in movies in SwaggerUI: [
  "string"
]